### PR TITLE
fix(ssh,security): strip synthetic newline and fix redirect target detection

### LINKED ISSUE #5: Косметика и грубые эвристики
### a. Лишняя пустая строка в выводе каждой команды
**Где:** `ssh.rs`, формирование payload и извлечение `output` в `drain_until_marker_with_exit`.

**Проблема:** payload использует `printf '\n{tag}__%d__\n'` — ведущий `\n` добавляет
лишнюю пустую строку поверх собственного перевода строки команды. Тесты проходят из-за
`.trim()`, но в UI каждая команда получает хвостовой пустой ряд.

**Фикс:** ведущий `\n` в printf нужен для надёжного детекта начала строки-маркера —
оставь его, но при извлечении вывода сними ровно **один** хвостовой `\n`, который этот
printf добавил:
```rust
let output = buf[..line_start].to_string();
let output = output.strip_suffix('\n').map(str::to_string).unwrap_or(output);
```
(Снимаем строго один перевод строки — синтетический от printf, — не трогая остальной вывод.)

**Done:** `run("echo hi")` даёт `stdout == "hi\n"` (или `"hi"` — по твоему соглашению),
без второго пустого ряда. Подкорректируй существующие тесты, если они завязаны на формат.

### b. Грубый `writes_to_system_path`
**Где:** `crates/agent/src/security.rs`, функция `writes_to_system_path`.

**Проблема:** флагает любой `>`, за которым где-то *дальше в строке* встречается
системный путь. Пример ложного срабатывания: `grep x > /tmp/a; cat /etc/passwd`
посчитается записью в систему (срабатывание безопасное — лишнее подтверждение, — но
неточное).

**Фикс:** проверять не «где-то после `>`», а **непосредственно цель редиректа** —
токен сразу после `>` или `>>`. Набросок: найти оператор редиректа, взять следующий
непробельный токен, проверить, что он начинается с системного пути.

**Done:** `writes_to_system_path("echo foo > /etc/passwd") == true`,
`writes_to_system_path("grep x > /tmp/a; cat /etc/passwd") == false`. Добавь оба кейса
в юнит-тесты.

### PR DESCRIPTION
**Pull request by devlawey**
## Summary

Two fixes from issue #5:

### Part A: Extra empty line in command output (`ssh.rs`)

The printf marker uses a leading `\n` for reliable line-start detection. This `\n` was leaking into the extracted command output as an extra trailing empty line. Tests passed due to `.trim()`, but the UI showed a blank row after each command.

**Fix:** After extracting output from the buffer, strip exactly one trailing `\n` — the synthetic one from printf, not the command's own output.

### Part B: Crude `writes_to_system_path` (`security.rs`)

The function checked if any system path appears anywhere after a `>` character, causing false positives like `grep x > /tmp/a; cat /etc/passwd` (flagged because `/etc/` appears in the read part).

**Fix:** Rewritten to split by shell operators (`;`, `|`, `&`), find each `>` or `>>` redirect, and check only the **immediate next token** (the redirect target). `/dev/null` is excluded as a null device.

## Test cases

`detect_system_redirect` expanded to 7 cases:
- `echo foo > /etc/passwd` → true
- `echo bar >> /boot/grub.cfg` → true (append)
- `echo foo > /tmp/test` → false (non-system path)
- `echo foo > /dev/null` → false (null device)
- `echo foo 2>/dev/null` → false (stderr redirect)
- `grep x > /tmp/a; cat /etc/passwd` → false (system path in read, not redirect)
- `dd if=/dev/zero > /dev/sda` → true (real device)

## Test results

```
70 passed; 0 failed; 5 ignored (Docker SSH integration)
```

The 5 ignored tests require a Docker sshd container and should be verified manually.

Closes #5

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed command output so results no longer include an unintended extra trailing newline.
  * Improved redirect handling when checking whether commands write to sensitive system paths, reducing false positives across split shell segments and append redirects.
  * Ensured `/dev/null`-style exclusions and quoted redirect targets are handled correctly.
* **Tests**
  * Expanded unit coverage for redirect detection with additional edge cases (e.g., quoted targets, non-ASCII content, and device-path-like targets).
* **Documentation**
  * Updated progress notes with details on the above fixes.

### CODE REVIEW

File: crates/agent/src/security.rs [hunk 1]
```diff
@@ -79,24 +79,55 @@ pub fn is_destructive(command: &str) -> bool {
...
+                while target_start < chars.len() && chars[target_start].is_whitespace() {
+                    target_start += 1;
+                }
+                // Extract the target token (up to next whitespace or end).
+                let mut target_end = target_start;
+                while target_end < chars.len() && !chars[target_end].is_whitespace() {
+                    target_end += 1;
+                }
+                if target_end > target_start {
+                    let target: String = sub[target_start..target_end].to_string();
```
[the remark this answers is not in the document]
Comment (devlawey) [reply crates/agent/src/security.rs]: Fixed in commit 22d3b35.

Switched from `chars().collect()` to `char_indices().collect()`. Now `target_start`/`target_end` are indices into a `Vec<(usize, char)>`, and byte offsets are extracted via `indices[target_start].0` / `indices[target_end].0` for slicing. Added test case with Cyrillic text before `>` to verify no panic.

File: crates/agent/src/security.rs [hunk 2]
```diff
@@ -79,24 +79,55 @@ pub fn is_destructive(command: &str) -> bool {
...
+                    // /dev/null is a null device, not a real system path write.
+                    if target != "/dev/null" {
+                        if system_paths.iter().any(|p| target.starts_with(p)) {
```
[the remark this answers is not in the document]
Comment (devlawey) [reply crates/agent/src/security.rs:132]: Fixed in commit 22d3b35.

After extracting the target token, `trim_matches(|c| c == '"' || c == '\'')` strips surrounding quotes before checking against system paths. Added test cases for both `>"/etc/passwd"` and `>>'/etc/passwd'`.
